modules/trustxBidAdapter.js

```diff
@@ -0,0 +1,210 @@
+var utils = require('src/utils.js');
+var bidfactory = require('src/bidfactory.js');
+var bidmanager = require('src/bidmanager.js');
+var adloader = require('src/adloader');
+var adaptermanager = require('src/adaptermanager');
+
+var TrustxAdapter = function TrustxAdapter() {
+  var bidderCode = 'trustx';
+  var reqHost = '//sofia.trustx.org';
+  var reqPath = '/hb?';
+  var LOG_ERROR_MESS = {
+    noAuid: 'Bid from response has no auid parameter - ',
+    noAdm: 'Bid from response has no adm parameter - ',
+    noBid: 'Array of bid objects is empty',
+    noPlacementCode: 'Can\'t find placementCode for bid with auid - ',
+    havePCodeFor: ', placementCode is available only for the following uids - ',
+    emptyUids: 'Uids should be not empty',
+    emptySeatbid: 'Seatbid array from response has empty item',
+    emptyResponse: 'Response is empty',
+    hasEmptySeatbidArray: 'Response has empty seatbid array',
+    hasNoArrayOfBids: 'Seatbid from response has no array of bid objects - '
+  };
+
+  function _makeHandler(auids, placementMap, onTimeoutWorker, needToSendStat) {
+    var cbName = bidderCode + '_callback_wrapper_' + auids.join('_');
+    $$PREBID_GLOBAL$$[cbName] = function(resp) {
+      delete $$PREBID_GLOBAL$$[cbName];
+      _responseProcessing(resp, auids, placementMap, onTimeoutWorker, needToSendStat);
+    };
+    return '$$PREBID_GLOBAL$$.' + cbName;
+  }
+
+  function _sendRequest(auids, placementMap, onTimeoutWorker) {
+    var query = [];
+    var path = reqPath;
+    var needToSendStat = Math.random() < 0.1;
+    query.push('u=' + encodeURIComponent(location.href));
+    query.push('auids=' + encodeURIComponent(auids.join(',')));
+    query.push('cb=' + _makeHandler(auids, placementMap, onTimeoutWorker, needToSendStat));
+    query.push('pt=' + (window.globalPrebidTrustxPriceType === 'net' ? 'net' : 'gross'));
+    query.push('hbstat=' + (needToSendStat ? '1' : '0'));
+
+    adloader.loadScript(reqHost + path + query.join('&'));
+  }
+
+  function _makeTimeoutWorker() {
+    var active = true;
+    var handler = function() {
+      worker.called = true;
+      $$PREBID_GLOBAL$$.offEvent('bidTimeout', handler);
```

Comment:
`$$PREBID_GLOBAL$$.offEvent` and `$$PREBID_GLOBAL$$.onEvent` are meant for publisher use and shouldn't be needed within bidder adapters

Reply:
Could you please let us know what we should replace $$PREBID_GLOBAL$$.offEvent and $$PREBID_GLOBAL$$.onEvent with?

Reply:
Could you please let us know what we should replace $$PREBID_GLOBAL$$.offEvent and $$PREBID_GLOBAL$$.onEvent with?